### PR TITLE
Warnings

### PR DESCRIPTION
**Pull request by sadrasabouri**
#### Reference Issues/PRs
#134 
#### What does this implement/fix? Explain your changes.
some warnings and errors added.

### CODE REVIEW

File: test/error_test.py [hunk 1]
```diff
@@ -34,10 +34,28 @@
...
 >>> g = GenerativeImage(lambda x,y: x, lambda x,y: y)
 >>> result = g.save_config()
 Traceback (most recent call last):
         ...
 samila.errors.samilaConfigError: Config file can't be saved. At least one of the function1_str or function2_str is None.
+>>> from samila.functions import *
+>>> select_color(2)
+Traceback (most recent call last):
+        ...
+samila.errors.samilaPlotError: Given color type is not supported. Replacing it with the default color.
```
Comment (sepandhaghighi) [inline test/error_test.py:51]: What?!
Replacing it with the default color?
It's an error!

File: samila/params.py [hunk 2]
```diff
@@ -39,8 +39,11 @@
 CONFIG_NO_STR_FUNCTION_ERROR = "Config file can't be saved. At least one of the function1_str or function2_str is None."
 PLOT_DATA_ERROR = "Plotting process can't be Done because data{0} is empty. Use generate method first."
 SAVE_NO_DATA_ERROR = "Data file can't be saved. At least one of the data1 or data2 is None."
+INVALID_COLOR_TYPE_ERROR = "Given color type is not supported. Replacing it with the default color."
```
Comment (sepandhaghighi) [inline samila/params.py:42]: Sometimes it's not the default color, it's the previously selected color!

Comment (sepandhaghighi) [reply samila/params.py:42]: My suggestion:

Given color type is not supported. It is set to the previously selected or default one.

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]

File: test/error_test.py (hunk 1 above)
Comment (sadrasabouri) [reply test/error_test.py:51]: Yes, you're right.

File: samila/params.py [hunk 3]
```diff
@@ -39,8 +39,11 @@
 CONFIG_NO_STR_FUNCTION_ERROR = "Config file can't be saved. At least one of the function1_str or function2_str is None."
 PLOT_DATA_ERROR = "Plotting process can't be Done because data{0} is empty. Use generate method first."
 SAVE_NO_DATA_ERROR = "Data file can't be saved. At least one of the data1 or data2 is None."
+INVALID_COLOR_TYPE_ERROR = "Given color type is not supported."
 MATPLOTLIB_VERSION_WARNING = "Source matplotlib version({0}) is different from yours, plots may be different."
 CALCULATION_EXCEPTION_WARNING = "The given functions are undefined at some points. Your plot may not be complete."
+BOTH_COLOR_COMPLEMENT_WARNING = "Both color and bgcolor are 'complement'. Both are set to default."
```
Comment (sepandhaghighi) [inline samila/params.py:45]: My suggestion:

It is not possible to set color and bgcolor to "complement" at the same time! Both are automatically set to the previous or default selection."

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]

**Review by sepandhaghighi — APPROVED** [review]
LGTM 🚀

**Pull request merged**